src/xcp_ng_dev/cli.py

```diff
@@ -37,67 +39,111 @@ def is_podman(runner):
+    parser = argparse.ArgumentParser()
+    subparsers_env = parser.add_subparsers(required=True, title="Development environments",
+        help="Available environments")
+
+    # container-based workflow
+    parser_container = subparsers_env.add_parser('container', help="Use a local container to build a package")
+    parser_container.set_defaults(func=container)
+    subparsers_container = parser_container.add_subparsers(dest='action',
+        help="Actions available for developing packages")
+
+    # build -- build an rpm using a container
+    parser_build = argparse.ArgumentParser()
+    parser_build = subparsers_container.add_parser('build',
+        help="Install dependencies for the spec file(s) found in the SPECS/ subdirectory "
+             "of the directory passed as parameter, then build the RPM(s). "
+             "Built RPMs and SRPMs will be in RPMS/ and SRPMS/ subdirectories. "
+             "Any preexisting BUILD, BUILDROOT, RPMS or SRPMS directories will be removed first.")
+    parser_build.add_argument('build_local', nargs='?', default='.',
+        help="Root path where SPECS/ and SOURCES are available. "
+             "The default is the working directory")
+    parser_build.add_argument('--define',
+        help="Definitions to be passed to rpmbuild. Example: --define "
+             "'xcp_ng_section extras', for building the 'extras' "
+             "version of a package which exists in both 'base' and 'extras' versions.")
+    parser_build.add_argument('-o', '--output-dir',
+        help="Directory where the RPMs, SRPMs and the build logs will appear. "
+             "The directory is created if it doesn't exist")
+    parser_build.add_argument('--rpmbuild-opts', action='append',
+        help="Pass additional option(s) to rpmbuild")
+    parser_build.add_argument('--rpmbuild-stage', action='store',
+        help=f"Request given -bX stage rpmbuild, X in [{RPMBUILD_STAGES}]")
+    add_container_args(parser_build)
+    add_common_args(parser_build)
+
+    # run -- execute commands inside a container
+    parser_run = subparsers_container.add_parser('run',
+                        help='Execute a command inside a container')
+    parser_run.add_argument('command', nargs=argparse.REMAINDER,
+                        help='Command to run inside the prepared container')
+    add_container_args(parser_run)
+    add_common_args(parser_run)
+
+    # shell -- like run bash
+    parser_shell = argparse.ArgumentParser()
+    parser_shell = subparsers_container.add_parser('shell',
+                        help='Drop a shell into the prepared container')
+    add_container_args(parser_shell)
+    add_common_args(parser_shell)
+
+    return parser
+
+def container(args):
+    build = False
+
+    if args.action == 'build':
+        build = True
+    elif args.action == 'run':
+        pass
+    elif args.action == 'shell':
+        pass
+    else:
+        print(f'Unknown action: {args.action}, exiting')
+        sys.exit(1)
```

Comment:
The experience could be improved (list possible choices?  can't argparse take care of this by itself?):
```
user@work-vates:build-env (dev/pau/package =)$ xcp-ng-dev container
Unknown action: None, exiting
```

Reply:
and yet...
```
$ xcp-ng-dev container foo
usage: xcp-ng-dev container [-h] {build,run,shell} ...
xcp-ng-dev container: error: argument action: invalid choice: 'foo' (choose from 'build', 'run', 'shell')
```
>_>